Fix the swizzle2d traversal order

lancerts opened the pull request:
## PR Description
Fix #4008 

The proposal in the issue is indeed correct.
```
new_i = off_i + ((ij % size_gj) % size_g)
new_j = (ij % size_gj) // size_g
```

Details
```
# linear index with respect to the first element in this group
ij = ij % size_gj
# new row and column indices
new_i = off_i + ij % size_g    -> row index (ij % size_g) wrt the first element in this group 
                                             -> plus the row index of the first element itself.
new_j = ij // size_g               -> column index (ij // size_g) wrt the first element in this group
```


## Tests 
Added `test_swizzle2d` in `/python/test/unit/language/test_standard.py`.

## Pre-commit 
checks passed

jlebar commented:
Thank you for the PR.

I added a PR template earlier this week -- which it looks like you saw and then deleted -- which asks for tests.  I added it because it's getting a little repetitive to ask so many PR authors to add tests.  Without accusing you in particular, would you be willing to help me understand how I could change the PR template in a way that might have caused you to write a test, or explain why testing isn't applicable?

lancerts commented:
> Thank you for the PR.
> 
> I added a PR template earlier this week -- which it looks like you saw and then deleted -- which asks for tests. I added it because it's getting a little repetitive to ask so many PR authors to add tests. Without accusing you in particular, would you be willing to help me understand how I could change the PR template in a way that might have caused you to write a test, or explain why testing isn't applicable?

Thanks for the comments. The PR template is very helpful indeed. 
Wrt the test section,
```
- [ ] **Tests** have been added and/or updated.
  - For changes to the backend: `/test/` (for `lit`), `/unittest/` (for
    `gtest`), or occasionally end-to-end tests like in
    `/python/test/unit/language/test_core.py`.
  - For changes to the frontend: `/python/test/`   
```
Seems the backend and frontend both involve `/python/test/`, which may cause confusion. 
In my case, I was planning to add a test in `/python/test/unit/language/test_standard.py`. 
But due to `For changes to the frontend: `/python/test/` ` holds me back due to being afraid of adding the test in the wrong place.

Suggestion on the template:
```
- [ ] **Tests** have been added and/or updated. This is MANDATORY.  
  - For changes to the backend: `/test/` (for `lit`), `/unittest/` (for
    `gtest`), or occasionally end-to-end tests like in
    `/python/test/unit/language/test_core.py`.
  - For changes to the frontend: `/python/test/`  
Note, both the backend and frontend have tests in `/python/test/`. Please locate the according test file based on the change.
```

lancerts commented:
Based on the template and feedback, added test_swizzle2d in /python/test/unit/language/test_standard.py. (can move to a different location if this is not the right place for this test case). Thanks

jlebar commented:
@Jokeren do you know if this PR is correct?

Jokeren commented:
Yes, what proposed in the original issue is correct since the function explicitly specifies *col-major for each group*.

More fixes to the function comment:

```
Transforms the indices of a row-major `size_i * size_j` matrix into the indices of a column-major matrix for each group of `size_g` rows.
```

However, the update to the tutorial is not necessary since the correctness doesn't depend on the order of the last group. It's correct as long as their pids are unique. I'm OK with the changes though because we have seen many users asking questions. WDYT @jlebar

jlebar commented:
> WDYT @jlebar

Happy to defer to you, Keren.

lancerts commented:
> Yes, what proposed in the original issue is correct since the function explicitly specifies _col-major for each group_.
> 
> More fixes to the function comment:
> 
> ```
> Transforms the indices of a row-major `size_i * size_j` matrix into the indices of a column-major matrix for each group of `size_g` rows.
> ```
> 
> However, the update to the tutorial is not necessary since the correctness doesn't depend on the order of the last group. It's correct as long as their pids are unique. I'm OK with the changes though because we have seen many users asking questions. WDYT @jlebar

Thanks for the suggestions on the function comment, updated it accordingly. 
Agreed that it does not impact the correctness of the tutorial and hence can remove the changes.
But just want to make sure things are consistent.
LMK whether i should just remove the changes in the tutorial or keep it as is. Thanks
